### PR TITLE
【Hackathon No.21】

### PR DESCRIPTION
**Pull request by yangguohao**
add rfc of SoftMarginLoss

### CODE REVIEW

File: rfcs/APIs/20220416_api_design_for_SotftMarginLoss.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+# 四、对比分析
+两种方案对比：
+- 采用pytorch方法，对算子进行CPU以及GPU设计
+
+# 五、设计思路与实现方案
+paddle.nn.SoftMarginLoss( input: Tensor, target: Tensor, reduction(str,可选) ) -> Tensor:
+
+paddle.nn.functional.soft_margin_loss(input, target, reduction: str = "mean", name:str=None, ) -> Tensor:
+input:Tensor, 维度为[batchsize,num_classes]
+label:Tensor, 维度为[batchsize,num_classes]
```
Comment (yingyibiao) [inline rfcs/APIs/20220416_api_design_for_SotftMarginLoss.md:77]: 1. input 和 label 的维度设置 [batch_size,dim] 是基于什么考虑呢？
2. input 和 label 接受的 data_type 是什么样子的呢？

Comment (yangguohao) [reply rfcs/APIs/20220416_api_design_for_SotftMarginLoss.md]: 已修改

File: rfcs/APIs/20220416_api_design_for_SotftMarginLoss.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+两种方案对比：
+- 采用pytorch方法，对算子进行CPU以及GPU设计
+
+# 五、设计思路与实现方案
+## 命名与参数设计
+- paddle.nn.SoftMarginLoss(reduction(str,可选)，name(str，可选)) -> Tensor:
+
+- paddle.nn.functional.soft_margin_loss(input, target, reduction: str = "mean", name:str=None, ) -> Tensor:
+    - input:Tensor, 维度为[N,*],其中N是batch_size， `*` 是任意其他维度。数据类型是float32、float64。
+    - label:Tensor, 维度为[batchsize,num_classes]维度、数据类型与输入 input 相同。
```
Comment (yingyibiao) [inline rfcs/APIs/20220416_api_design_for_SotftMarginLoss.md:78]: 这里的 API 参数设计还存在一些问题：
1. label 的维度应该与input的维度相同
2. label 的数值只能为 1 或者 -1
3. label 需要支持 [int32, int64, float32, float64] 数据类型

Comment (yangguohao) [reply rfcs/APIs/20220416_api_design_for_SotftMarginLoss.md]: 已修改

**Review by yingyibiao — APPROVED** [review]
